src/opensearch.py

```diff
@@ -88,7 +88,7 @@ def is_service_started(self, paused: Optional[bool] = False) -> bool:
         # Now, we must dig deeper into the actual status of systemd and the JVM process.
         # First, we want to make sure the process is not stopped, dead or zombie.
         try:
-            pid = run_cmd("lsof", args="-ti:9200").rstrip()
+            pid = run_cmd("lsof", args="-ti:9200").out.rstrip()
```

Comment:
Can you explain this? `run_cmd` returns `output.stdout.strip()` - there is no `.out` and no need for `.rstrip()`

Reply:
@Mehdi-Bendriss I think we are confusing `OpenSearchDistribution.{run_bin,run_script}` and `from charms.opensearch.v0.helper_charm import run_cmd`. Indeed, the former just provide a `str`. However, they are not generic cmd calls, they are based on `OPENSEARCH_BIN` and `OPENSEARCH_HOME` folders. We want something in the `PATH` (`lsof`). The latter gives this option and returns a `SimpleNamespace`.

I did not notice myself the slight distinction, as we do have `OpenSearchDistribution._run_cmd` as well. I think the `run_cmd` was added by @reneradoi. I will set a quick chat between us tomorrow to go over which is the most appropriate.

Reply:
@Mehdi-Bendriss I will summarize our offline discussion with @reneradoi:
1) Continue with this implementation
2) Eventually, we will make `OpenSearchDistribution._run_cmd` obsolete

Reply:
Reported as a bug: https://github.com/canonical/opensearch-operator/issues/489